### PR TITLE
Update year in the notice file and the release process instructions

### PR DESCRIPTION
**Pull request by abhishekagarwal87**

### CODE REVIEW

File: distribution/asf-release-process-guide.md [hunk 1]
```diff
@@ -71,6 +71,17 @@ Finally, the sample [`docker-compose.yml`](https://github.com/apache/druid/blob/
 
 Once this is completed, open a PR to the master branch. Also, be sure to confirm that these versions are all correct in the release branch, otherwise fix them and open a backport PR to the release branch.
 
+#### Version updates on master and release branch
+
+You also need to update the version manually in some places for both master branch and the release branch.
+- `DRUID_DOCS_VERSION` has to be changed to `0.17.0` in `web-console/src/links.ts`.
+- Docker image tag has to be changed to `0.17.0` for all druid services in `distribution/docker/docker-compose.yml`.
```
Comment (clintropolis) [inline distribution/asf-release-process-guide.md:78]: this part about updating docker-compose.yml is already mentioned https://github.com/apache/druid/blob/master/distribution/asf-release-process-guide.md?plain=1#L62

File: distribution/asf-release-process-guide.md [hunk 2]
```diff
@@ -71,6 +71,17 @@ Finally, the sample [`docker-compose.yml`](https://github.com/apache/druid/blob/
 
 Once this is completed, open a PR to the master branch. Also, be sure to confirm that these versions are all correct in the release branch, otherwise fix them and open a backport PR to the release branch.
 
+#### Version updates on master and release branch
```
Comment (clintropolis) [inline distribution/asf-release-process-guide.md:74]: I think these can all be done in the master branch and be part of the previous section, maybe instead add a bit that if any of these steps is missed, then update it in master and backport to the release branch?

File: distribution/asf-release-process-guide.md [hunk 3]
```diff
@@ -248,10 +259,16 @@ must be tagged properly to make this script working. See the above [Release note
 Once the release branch is good for an RC, you can build a new tag with:
 
 ```bash
-$ mvn -DreleaseVersion=0.17.0 -DdevelopmentVersion=0.18.0-SNAPSHOT -Dtag=druid-0.17.0-rc3 -DpushChanges=false clean release:clean release:prepare
+$ mvn -Pwebsite-docs -DreleaseVersion=0.17.0 -DdevelopmentVersion=0.18.0-SNAPSHOT -Dtag=druid-0.17.0-rc3 -DpushChanges=false clean release:clean release:prepare
```
Comment (clintropolis) [inline distribution/asf-release-process-guide.md:262]: does this do the right thing? The website-docs profile only has the website module, so will it only bump website version or is it additive?

Comment (abhishekagarwal87) [reply distribution/asf-release-process-guide.md:262]: yeah. I verified it by running `mvn -Pwebsite-docs help:evaluate -Dexpression=project.modules`

**Pull request merged**